src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/domainEvents/ComAllocatedHandler.kt

```diff
@@ -9,68 +9,134 @@ import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.model.request.Updat
 import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.OffenderService
 import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.StaffService
 import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.probation.DeliusApiClient
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.probation.WorkLoadApiClient
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.probation.mapper.OffenderManagerMapper
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.probation.model.OffenderManager
 
 @Service
 class ComAllocatedHandler(
   private val deliusApiClient: DeliusApiClient,
+  private val workLoadApiClient: WorkLoadApiClient,
   private val offenderService: OffenderService,
   private val objectMapper: ObjectMapper,
   private val staffService: StaffService,
+  private val offenderManagerMapper: OffenderManagerMapper,
 ) {
   private companion object {
     val log: Logger = LoggerFactory.getLogger(this::class.java)
   }
 
   fun handleEvent(message: String) {
-    val event = objectMapper.readValue(message, HMPPSDomainEvent::class.java)
-    val crn = event.personReference.crn()
-    if (crn != null) {
-      processComAllocation(crn)
-    }
-  }
+    log.info("Received COM allocation event")
 
-  fun processComAllocation(crn: String) {
-    log.info("processing COM allocation for CRN $crn")
-    val offenderManager = deliusApiClient.getOffenderManager(crn)
-    if (offenderManager == null) {
-      log.warn("Could not find offender manager for CRN $crn")
+    val event = try {
+      objectMapper.readValue(message, HMPPSDomainEvent::class.java)
+    } catch (e: Exception) {
+      log.error("Failed to parse event message", e)
       return
     }
 
-    // If the COM does not have a username, they are assumed to be ineligible for use of this service. (e.g. the "unallocated" staff members)
-    log.info("responsible offender manager code for crn $crn is ${offenderManager.code}")
-    if (offenderManager.username == null) {
-      log.warn("offender manager with code ${offenderManager.code} does not have a username")
+    val crn = event.personReference.crn()
+    if (crn.isNullOrBlank()) {
```

Comment:
why would this ever be null or blank?

Reply:
have  no idea,.. I assumed it was there for a reason

Reply:
will remove then